src/braket/default_simulator/cirq/program_context.py

```diff
@@ -0,0 +1,88 @@
+from typing import Tuple, List
+
+import cirq
+import numpy as np
+from braket.ir.jaqcd.program_v1 import Results
+from cirq import Circuit
+
+from braket.default_simulator.openqasm.program_context import AbstractProgramContext
+
+CIRQ_GATES = {
+    "i": cirq.I,
+    "h": cirq.H,
+    "x": cirq.X,
+    "y": cirq.Y,
+    "z": cirq.Z,
+    "cnot": cirq.CNOT,
+    "cz": cirq.CZ,
+    "s": cirq.S,
+    "t": cirq.T,
+    "cphaseshift": cirq.cphase,
+    "rx": cirq.rx,
+    "ry": cirq.ry,
+    "rz": cirq.rz,
+    "swap": cirq.SWAP,
+    "iswap": cirq.ISWAP,
+    # "xx": cirq.XX,
+    # "yy": cirq.YY,
+    # "zz": cirq.ZZ,
+    "ccnot": cirq.CCNOT,
+    "cswap": cirq.CSWAP,
+    "ccz": cirq.CCZ,
+    "ccx": cirq.CCX,
+    "measure": cirq.MeasurementGate,
+}
+
+
+class ProgramContext(AbstractProgramContext):
+    def __init__(self):
+        super().__init__(Circuit())
+
+    def is_builtin_gate(self, name: str) -> bool:
+        user_defined_gate = self.is_user_defined_gate(name)
+        return name in CIRQ_GATES and not user_defined_gate
+
+    def add_phase_instruction(self, target: Tuple[int], phase_value: int):
+        raise NotImplementedError
+
+    def _gate_accepts_parameters(self, gate_class):
+        return hasattr(gate_class, "parameters") and len(gate_class.parameters) > 0
+
+    def add_gate_instruction(
+        self, gate_name: str, target: Tuple[int], params, ctrl_modifiers: List[int], power: int
+    ):
+        qubits = [cirq.LineQubit(int(qubit)) for qubit in target]
+        target_qubits = qubits[len(ctrl_modifiers) :]
+        control_qubits = qubits[: len(ctrl_modifiers)]
+        try:
+            if params:
+                gate = CIRQ_GATES[gate_name](*params).on(*target_qubits)
```

Comment:
When the gates accept an angle parameter. Few gates were already initialized. I was using them. I could use on in both the cases.

Let me change it quickly to avoid confusion